shell/shell.go

```diff
@@ -47,7 +48,12 @@ func ExecuteShell(uid uint32,
 		cmd.SysProcAttr.Credential = &syscall.Credential{Uid: uid, Gid: gid}
 	}
 
+	if _, err := os.Stat(homeDir); !os.IsNotExist(err) {
+		cmd.Dir = homeDir
+	}
+	cmd.Env = append(cmd.Env, fmt.Sprintf("HOME=%s", homeDir))
```

Comment:
Is it worth considering using something like `su -l <USER>` to make it a true login shell? I'm not sure exactly what all the differences are, but I know `su` does some extra things, like setting up the environment and putting the shell in its own process group. And specifically related to this PR, it sets the home directory.

Reply:
@kacf can we expect to have `su -l user` everywhere, in all yocto environment et al?

Reply:
Great, thanks for the confirmation.

Reply:
@kacf one thing, though: if we do it this way, we have to run mender-connect as root, don't we?

Reply:
You're right. I thought it was possible to run it password-less if you su to the same user as the one running mender-connect, but this doesn't work. In that case I agree, better to stick to what we have for now.